modules/ismp/clients/sync-committee/src/beacon_client.rs

```diff
@@ -25,211 +30,385 @@ use crate::{
+			match consensus_mechanic {
+				L2Consensus::ArbitrumOrbit(rollup_core_address) => {
+					let l2_client =
+						ArbitrumOrbitClient::<H>::new(state_machine.clone(), *rollup_core_address);
+					client.register_l2_client(Box::new(l2_client));
+				},
+				L2Consensus::ArbitrumBold(rollup_core_address) => {
+					let l2_client =
+						ArbitrumBoldClient::<H>::new(state_machine.clone(), *rollup_core_address);
+					client.register_l2_client(Box::new(l2_client));
+				},
+				L2Consensus::OpL2Oracle(l2_oracle) => {
+					let l2_client =
+						OptimismL2OracleClient::<H>::new(state_machine.clone(), *l2_oracle);
+					client.register_l2_client(Box::new(l2_client));
+				},
+				L2Consensus::OpFaultProofs((dispute_game_factory, respected_game_type)) => {
+					let l2_client = OptimismFaultProofClient::<H>::new(
+						state_machine.clone(),
+						*dispute_game_factory,
+						vec![*respected_game_type],
+					);
+					client.register_l2_client(Box::new(l2_client));
+				},
+				L2Consensus::OpFaultProofGames((dispute_game_factory, respected_game_types)) => {
+					let l2_client = OptimismFaultProofClient::<H>::new(
+						state_machine.clone(),
+						*dispute_game_factory,
+						respected_game_types.clone(),
+					);
+					client.register_l2_client(Box::new(l2_client));
+				},
+			}
+		}
+
+		Ok(client)
 	}
 }
 
+// Updated implementation of the ConsensusClient trait
 impl<
 		H: IsmpHost + Send + Sync + Default + 'static,
 		C: Config + Send + Sync + Default + 'static,
-		I: 'static,
 		T: pallet_ismp_host_executive::Config + pallet::Config<I> + 'static,
-	> Clone for SyncCommitteeConsensusClient<H, C, T, I>
+		I: 'static,
+	> ConsensusClient for SyncCommitteeConsensusClient<H, C, T, I>
 {
-	fn clone(&self) -> Self {
-		Self(core::marker::PhantomData)
+	fn verify_consensus(
+		&self,
+		host: &dyn IsmpHost,
+		consensus_state_id: ConsensusStateId,
+		trusted_consensus_state: Vec<u8>,
+		consensus_proof: Vec<u8>,
+	) -> Result<(Vec<u8>, VerifiedCommitments), Error> {
+		let consensus_state = ConsensusState::decode(&mut &trusted_consensus_state[..])
+			.map_err(|_| Error::Custom("Cannot decode trusted consensus state".to_string()))?;
+
+		// Parse update based on the type
+		let update = match BeaconClientUpdate::decode(&mut &consensus_proof[..]) {
+			Ok(update) => update,
+			Err(_) => {
+				// Try to decode as a standalone L2 update
+				match L2StateUpdate::decode(&mut &consensus_proof[..]) {
+					Ok(l2_update) => {
+						// Handle standalone L2 update without changing beacon state
+						return self.process_l2_update(
+							host,
+							consensus_state_id,
+							consensus_state,
+							l2_update,
+						);
+					},
+					Err(_) => {
+						return Err(Error::Custom("Cannot decode update".to_string()));
+					},
+				}
+			},
+		};
+
+		// Process full beacon update with optional L2 updates
+		self.process_beacon_update(host, consensus_state_id, consensus_state, update)
+	}
```

Comment:
The goal is that the verification of L2 state assertions to happen independent of the sync-committee client so it can be updated independently, potentially replacing it entirely with our future zk-casper light client. So you'd have to put the verifiers into new crates